### ISSUE TITLE
Networking tool disassociation does not roll back configuration modes

### ISSUE THREAD

**Issue by bakabaka** [xchwarze/wifi-pineapple-cloner#16]
Using the UI in the association tool under the networking tab you can connect properly to an access point and works really well.
A behavior was noticed: disassociating from an access point that is sharing network access to the pineapple leaves the pineapple unusable due to the fact that the configuration modes in the /etc/config/wireless config file apparently leave the blocks in "sta" mode instead of rolling back to "ap" mode. This happens inside the wifi-iface blocks.

**Comment by xchwarze** [comment]
If you find a solution to this, could you send the PR to https://github.com/xchwarze/wifi-pineapple-panel

**Comment by bakabaka** [comment]
Granted, im still looking.

**Comment by bakabaka** [comment]
imagine my face when trying to build a debug image to troubleshoot the issues i have stumbled upon. and this happened:
<img width="1367" alt="image" src="https://user-images.githubusercontent.com/207563/190014822-0dd10504-e3c9-49fd-96a2-a886d352807c.png">

yeah, i have to use an x86_64 vm...

In any case the issues seem to come from the init scripts. so i guess we could call this issue closed(?)

**Comment by xchwarze** [comment]
The original pineapple is designed to never use the wlan0 adapter (never use it for anything other than being connected to it).
The solution they found in the mk7 was to add an extra wifi adapter from the factory.
You could also modify some things in the panel to support this new use case, although this would not be as robust as having 3 Wi-Fi cards installed.

**Comment by bakabaka** [comment]
My setup is 2 rt5370's a 4 port usb hub, a 32GB nano usb, and the dual antena setup.

I came to the same conclusion although im tinkering with wlan0 and the files that point to it. In the meantime this can be closed.

After some toiling i implemented a webshell via ttyd and a small module to access it via the panels, i have not checked the panel repo at length but i think is nice to have. im attaching a few pictures of it. I have not dared to push anything since im new to the gitsphere as well but im working on this and can share if it is useful to you of course it needs improvement. Along with said shell; i implemented the left and right button support. right now it sets up the config files and reboot depending on where the switch is: left default conf, right "attack mode", this is useful for my AR150 as it only has the internal antenas. the attack configs are: setup the wireless station mode for wlan0 and setup a hotspot wlan0-1 (the button eliminates the need of an interface to setup), this can be further improved as the recent versions of the firmware implement "campaigns", we can implement profiles to have a setup like "left switch": rollback configs/reset firmware/panic reflash. "right switch": setup profiles selected on the ui (allowing for more than 2 "profiles"). i have been quite busy. currently im stuck trying to compile a driver of a wireless dongle i have the sources for and no experience compiling it for wrt. if you could lend a hand i'd be very grateful as this dongles seem to work nicely on my raspberry pi zero.

TTYD
<img width="836" alt="image" src="https://user-images.githubusercontent.com/207563/191869348-b0899137-f79b-42b2-885f-e86cc6b00efd.png">

Functions necesary for the switch to work on the button_script
<img width="769" alt="image" src="https://user-images.githubusercontent.com/207563/191869554-ec782fda-1bab-4e32-a975-c31773205cb6.png">

The button script i intend to weave into a module for profiling.
<img width="742" alt="image" src="https://user-images.githubusercontent.com/207563/191869705-79f9e97b-7c2b-4e4b-8f14-f18bcc439502.png">

Glamour shot
![image](https://user-images.githubusercontent.com/207563/191869902-8e32d772-1c21-4fd5-86e8-322b6c2ae5b2.png)

Not my personal blog. i know sorry.

**Comment by xchwarze** [comment]
> My setup is 2 rt5370's a 4 port usb hub, a 32GB nano usb, and the dual antena setup.

I have a similar setup but with two USB hubs that I alternate (one is powered, I use that to be able to use high-power Wi-Fi cards)
In your case, I would consider using the GL-AR750S, which is much more robust for such an extreme use case.
I used the Archer C7 but it's not very portable really...

> After some toiling i implemented a webshell via ttyd and a small module to access it via the panels, i have not checked the panel repo

I thought about doing something similar and being able to have the same shell as the MK7 has, but I still couldn't find time to experiment with it.
My idea was to do it as an extension and make a repo with different plugins that were taken and were never accepted in the official plugins repo.
I guess they're upset because I made this project xD

I think the best thing to do about the original problem is:

1. Turn things on and off based on the number of Wi-Fi cards connected. For example, which wifi is enabled to use it to connect to the internet.
In order not to have to do this, the MK7 has the 3 Wi-Fi but I think that for this project it would be an interesting feature.

2. Save a copy of certain configs to be able to use them every time we play with the wifi.

With these two changes, the connection with the pineapple would no longer be lost in any case and it would be much more robust.
It also seems to me that it is the easiest and fastest to develop and that it would run well on any router that is running this project
